openmdao/jacobians/jacobian.py

```diff
@@ -63,10 +62,20 @@ def _get_abs_key(self, key):
         try:
             return self._abs_keys[key]
         except KeyError:
-            abskey = key2abs_key(self._system(), key)
-            if abskey is not None:
+            resolver = self._system()._resolver
+            of = resolver.any2abs(key[0], 'output')
+            wrt = resolver.rel2abs(key[1], check=True)
```

Comment:
This step feels a lot more complicated now.

Reply:
You're definitely right about that.  It's similar now to the previous version.